src/scripts/main.js

```diff
@@ -1,3 +1,391 @@
+  const sortTody = document.querySelector('tbody');
+  const rows = [...sortTody.querySelectorAll('tr')];
+  const value = e.target.closest('th').textContent.toLowerCase();
+
+  const intex = e.target.cellIndex;
+  const cells = [];
+
+  sortTody.querySelectorAll('tr').forEach((item) => {
+    cells.push(item.cells[intex].textContent);
+  });
+
+  const newRows = rows.sort((a, b) => {
+    const cellsA = a.cells[intex].textContent;
+    const cellsB = b.cells[intex].textContent;
+
+    switch (value) {
+      case 'name':
+      case 'position':
+        return cellsA.localeCompare(cellsB);
+
+      case 'age':
+        return cellsA - cellsB;
+
+      case 'salary':
+        return (
+          +cellsA.replace(',', '').slice(1) - +cellsB.replace(',', '').slice(1)
+        );
+
+      default:
+        return null;
+    }
+  });
+
+  if (value === revers) {
+    newRows.reverse();
+    revers = '';
+  } else {
+    revers = value;
+  }
+
+  sortTody.innerHTML = '';
+
+  newRows.forEach((item) => {
+    sortTody.appendChild(item);
+  });
+});
+
+tbody.addEventListener('dblclick', (e) => {
+  const findInput = document.querySelector('.cell-input');
+  const oldValue = e.target.textContent;
+
+  if (findInput) {
+    findInput.remove();
+    e.target.textContent = oldValue;
+  }
+
+  const input = document.createElement('input');
+
+  input.setAttribute('class', 'cell-input');
+  e.target.textContent = '';
+  e.target.append(input);
+  input.focus();
+
+  if (
+    e.target === e.target.parentElement.children[3] ||
+    e.target === e.target.parentElement.children[4]
+  ) {
+    input.style.width = '60px';
+  } else {
+    input.style.width = '100px';
+  }
+
+  input.onkeydown = (even) => {
+    if (even.key === 'Enter') {
+      if (e.target === e.target.parentElement.children[3]) {
+        input.setAttribute('type', 'number');
+
+        if (input.value < 18 || input.value > 90) {
+          e.target.textContent = oldValue;
+          input.remove();
+
+          return;
+        }
+      }
+
+      if (e.target === e.target.parentElement.children[4]) {
+        input.setAttribute('type', 'number');
+
+        if (input.value <= 0) {
+          e.target.textContent = oldValue;
+          input.remove();
+
+          return;
+        } else {
+          e.target.textContent = `$${(+input.value).toLocaleString('en-US')}`;
+          input.remove();
+
+          return;
+        }
+      }
+
+      if (!input.value.trim()) {
+        e.target.textContent = oldValue;
+      } else {
+        e.target.textContent = input.value;
+      }
+
+      input.remove();
+    }
+  };
+
+  input.onblur = () => {
+    if (e.target === e.target.parentElement.children[3]) {
+      input.setAttribute('type', 'number');
+
+      if (input.value < 18 || input.value > 90) {
+        e.target.textContent = oldValue;
+        input.remove();
+
+        return;
+      }
+    }
+
+    if (e.target === e.target.parentElement.children[4]) {
```

Comment:
The same issue as mentioned for line 255 applies here in the 'onblur' event handler.